templates/vdbench_save.yml

```diff
@@ -0,0 +1,160 @@
+# Copyright (C) Microsoft Corporation. All rights reserved.
+# https://aka.ms/yaml
+
+steps:
+- script: |
+    export PATH=$PATH:${BUILD_SOURCESDIRECTORY}/test/utils
+
+    az login --service-principal -u $ARM_CLIENT_ID -p $ARM_CLIENT_SECRET --tenant $ARM_TENANT_ID &&
+    az account set -s $ARM_SUBSCRIPTION_ID
+    check_rc.sh $? "Login to Azure and set default subscription" || exit 0
+
+    # Wait for the vdbench.py script to complete.
+    cmdokmsg="CMD_OK_$(date -u '+%m%dx%H%M%S')"
+    cmdlog=/tmp/vm_runcmd.log
+    polling_secs=60
+    while [ 1 ]; do
+      az vmss run-command invoke                      \
+        --resource-group ${RESOURCE_GROUP_NAME}-vmss  \
+        --name           "vmss"                       \
+        --instance-id    0                            \
+        --command-id     RunShellScript               \
+        --scripts        "runuser -l azureuser -c 'tmux ls' || echo ${cmdokmsg}" \
+        --query 'value[0].message' --output tsv |& tee ${cmdlog}
+      check_rc.sh ${PIPESTATUS[0]} "Checking for any tmux session " || exit
+      grep -q ${cmdokmsg} ${cmdlog}
+      if [ $? -ne 0 ]; then
+        echo "INFO: tmux session still active. Re-check in ${polling_secs} seconds."
+        sleep ${polling_secs}
+        continue
+      else
+        echo "INFO: tmux session no longer active"
+        break
+      fi
+    done
+  displayName: "Check for vdbench.py completion"
```

Comment:
This is making the assumption that `lack of tmux session == vdbench success`. But in the (successful) test runs, was there some confirmation that vdbench succeeded? It'd be good to know if it ran but immediately failed (stored results are useless), ran X tests but failed at X+1 (stored results might have some value), or ran all X tests successfully (stored results are likely valuable).

For the results stored from testing this branch, did the results look OK/expected?

Reply:
Adding another check of to tail vdb.log and look for success message